### ISSUE TITLE
Dagger Engine should include everything needed for multi-arch builds: `binfmt`, QEMU

### ISSUE THREAD

**Issue by jpadams** [dagger/dagger#3801]
We should include `binfmt` in our Dagger Engine. This need was described in this issue https://github.com/dagger/dagger/issues/1371#issuecomment-1018407356 and has emerged in the de facto implementations of our multi-arch demos where we install `binfmt` for QEMU:
- https://github.com/kpenfound/greetings-api/blob/main/.circleci/config.yml#L11-L12
- https://github.com/kpenfound/greetings-api/blob/main/.github/workflows/test.yml#L11-L13
- https://gist.github.com/jpadams/ae659037469940fc5323ae8e79b59dbe#file-ci-for-ecs-demo-L7

**Comment by jpadams** [comment]
@sipsma @aluzzardi  what do you think?

**Comment by sipsma** [comment]
To clarify one thing, `binfmt` isn't just a package that needs to be installed, the `docker run --rm --privileged tonistiigi/binfmt:latest --install all` command actually results in the kernel being configured with qemu settings at runtime (basically registers emulators by writing to magic files under `/proc/sys`: https://docs.kernel.org/admin-guide/binfmt-misc.html).

We could make it so that our engine image does that automatically when it starts. My hesitation is that this is a kernel setting that's not namespaced to the container (afaik, apparently [there was a patch submitted to the kernel to namespace this](https://patchwork.kernel.org/project/linux-fsdevel/cover/20181010161430.11633-1-laurent@vivier.eu/) but I don't see that it was ever merged).

It's thus possible that changing this setting could thus interfere with something else users are running on the machine where the engine image is running. So even if we do run this by default we probably need a way to disable it, which means we need to find a way of exposing this knob to users when provisioning. In general we are trying to avoid any such knobs as much as possible...

So, tl;dr it's kind of complicated. I think for now I'd like to leave it as is unless it's becoming an active pain point for users.

**Comment by jpadams** [comment]
@sipsma says to leave as-is for now, so closing
